### PR TITLE
Jerryliu sc 69749 support signing modal on android

### PR DESCRIPTION
**Pull request by Ethella**
* Make activity context mandatory in all APIs
   * This would prevent enabling UI from Auth Relayer side crashes the app
*  Add @Keep to nested EIP712TypedData

### CODE REVIEW

**Comment by Ethella** [comment]
image

File: magic/ext/oauth/build.gradle [hunk 1]
```diff
@@ -37,8 +37,8 @@ dependencies {
     implementation 'androidx.browser:browser:1.4.0'
     implementation "com.google.code.gson:gson:$gson_version"
     implementation 'org.jetbrains.kotlin:kotlin-reflect:1.7.10'
-//    implementation project(':magic:core')
-    implementation 'link.magic:magic-android:[4.0,5.0['
+    implementation project(':magic:core')
+//    implementation 'link.magic:magic-android:[4.0,5.0['
```
Comment (Ariflo) [inline magic/ext/oauth/build.gradle]: nit: do we need to keep this commented out code?

Comment (Ethella) [reply magic/ext/oauth/build.gradle]: Good catch

**Review by Ariflo — APPROVED** [review]
LGTM. OOC, seeing as we are now directly setting the provider context [here](https://github.com/magiclabs/magic-android/pull/4/files#diff-8885785eba894a92c0c56cd94a375f8b92ee78ff8a4509e8b20e7f76b64e545cL14), is there any concern that it won't get passed to the overlay?

**Comment by Ethella** [comment]
> LGTM. OOC, seeing as we are now directly setting the provider context [here](https://github.com/magiclabs/magic-android/pull/4/files#diff-8885785eba894a92c0c56cd94a375f8b92ee78ff8a4509e8b20e7f76b64e545cL14), is there any concern that it won't get passed to the overlay?

"The overlay is a Webview widget set with internal access only, and cannot be accessed from external libraries such as our extension library. To allow for mutations, we provide the context property, but prevent access to the overlay from outside sources."

**Pull request merged**